### DISCUSSION TITLE
Ghost vertex and edges attached to vertex still there after deletion, possible stale index record of vertex and edge index

### DISCUSSION THREAD

**Discussion by sumandas0** [JanusGraph/janusgraph#3948] [Q&A] [answered]
Hello everyone,

I am encountering an issue with stale vertex edge indices in JanusGraph. I have deleted all attached vertices and edges from a JanusGraph vertex, but now whenever I search for edges from that vertex, I can still see them all, but they are empty. I had previously faced an issue with composite index and solved it with this [link](https://github.com/JanusGraph/janusgraph/issues/1099). However, this time, the problem lies with the vertex edge index being stale. Specifically, if you independently search by edge and vertex, you will not get anything, but if you just search by source vertex, you will see all the edges, where in reality there is nothing there, just wrong entries in the index.

Example:

```gremlin
gremlin> g.V(4423688264).bothE()
==>e[9i3e5uk-1hijzkw-qv4l-215qzpk][3235872992-inputs->4423688264]
==>e[9i3fod8-1hik928-qv4l-215qzpk][3235885280-inputs->4423688264]
==>e[9i3iknw-1hikov4-qv4l-215qzpk][3235905760-inputs->4423688264]
==>e[7mcwb2l-1m9afzc-qv4l-215qzpk][3522642024-inputs->4423688264]
==>e[7mcwqvh-1m9aj54-qv4l-215qzpk][3522646120-inputs->4423688264]
```
Even though all edges and vertices were deleted at once, I can still see them. I know that there is nothing there, just the wrong entry in the index. This is confirmed by checking their properties:

```gremlin
gremlin> g.V(4423688264).bothE().inV().valueMap()
==>[]
==>[]
==>[]
==>[]
==>[]
gremlin> g.E('9i3e5uk-1hijzkw-qv4l-215qzpk')
gremlin> g.E('9i3fod8-1hik928-qv4l-215qzpk')
gremlin> g.E('7mcwb2l-1m9afzc-qv4l-215qzpk')
gremlin> g.V(3235872992)
gremlin> g.V(3235885280)
gremlin> g.V(3522642024)
```
This is causing significant issues in our production instance.

JanusGraph version: 0.6
Storage Backend: Cassandra
Index Backend: Elasticsearch

Does anyone have any idea on how to resolve this issue? Any help would be greatly appreciated.

Thank you in advance.

**Comment by li-boxuan** [comment]
What happens when you do `g.V(4423688264).bothE().drop().iterate()`? Error?

**Reply by sumandas0** [reply]
So basically after I close the session it reappears again. It doesn't error out

**Reply by li-boxuan** [reply]
Even if the commit succeeded?

**Reply by sumandas0** [reply]
Yes

**Reply by li-boxuan** [reply]
A workaround would be to delete that vertex and add it back (together with normal edges).

**Reply by li-boxuan** [reply, marked as answer]
Another thing you could try: add a dummy property to that "half edge" and then try deleting it.
